code/go/0chain.net/blobbercore/handler/handler.go

```diff
@@ -648,7 +653,48 @@ func InsertShare(ctx context.Context, r *http.Request) (interface{}, error) {
 	return map[string]interface{}{"message": "Share info added successfully"}, nil
 }
 
-//PrintCSS - print the common css elements
+// ListShare a list of files that clientID has shared
+func ListShare(ctx context.Context, r *http.Request) (interface{}, error) {
+
+	ctx = setupHandlerContext(ctx, r)
+
+	var (
+		allocationID = ctx.Value(constants.ContextKeyAllocation).(string)
+		clientID     = ctx.Value(constants.ContextKeyClient).(string)
+	)
+
+	limit, err := common.GetOffsetLimitOrderParam(r.URL.Query())
+	if err != nil {
+		return nil, err
+	}
+
+	allocationObj, err := storageHandler.verifyAllocation(ctx, allocationID, true)
+	if err != nil {
+		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
+	}
+
+	sign := r.Header.Get(common.ClientSignatureHeader)
+
+	valid, err := verifySignatureFromRequest(allocationID, sign, allocationObj.OwnerPublicKey)
```

Comment:
what will happen if malicious blobber will ignore that check an will provide share info to everyone? can it be used maliciously?

Reply:
the output returned will have:
these fields:
```
owner_id
client_id
file_path_hash
re_encryption_key
client_encryption_public_key
revoked
expiry_at
available_at
```

but all these information are not useful without authToken.